cmd/bootkube/render.go

```diff
@@ -65,6 +71,12 @@ func init() {
 	cmdRender.Flags().StringVar(&renderOpts.etcdCAPath, "etcd-ca-path", "", "Path to an existing PEM encoded CA that will be used for TLS-enabled communication between the apiserver and etcd. Must be used in conjunction with --etcd-certificate-path and --etcd-private-key-path, and must have etcd configured to use TLS with matching secrets.")
 	cmdRender.Flags().StringVar(&renderOpts.etcdCertificatePath, "etcd-certificate-path", "", "Path to an existing certificate that will be used for TLS-enabled communication between the apiserver and etcd. Must be used in conjunction with --etcd-ca-path and --etcd-private-key-path, and must have etcd configured to use TLS with matching secrets.")
 	cmdRender.Flags().StringVar(&renderOpts.etcdPrivateKeyPath, "etcd-private-key-path", "", "Path to an existing private key that will be used for TLS-enabled communication between the apiserver and etcd. Must be used in conjunction with --etcd-ca-path and --etcd-certificate-path, and must have etcd configured to use TLS with matching secrets.")
+	cmdRender.Flags().StringVar(&renderOpts.selfHostedEtcdClientCAPath, "self-hosted-etcd-client-ca", "", "Path to an existing PEM encoded CA that will be used for self-hosted etcd's client port.")
+	cmdRender.Flags().StringVar(&renderOpts.selfHostedEtcdClientCertPath, "self-hosted-etcd-client-cert", "", "Path to an existing certificate that will be used for self-hosted etcd's client port.")
+	cmdRender.Flags().StringVar(&renderOpts.selfHostedEtcdClientKeyPath, "self-hosted-etcd-client-key", "", "Path to an existing private key that will be used for self-hosted etcd's client port.")
+	cmdRender.Flags().StringVar(&renderOpts.selfHostedEtcdPeerCAPath, "self-hosted-etcd-peer-ca", "", "Path to an existing PEM encoded CA that will be used for  self-hosted etcd's peer port.")
+	cmdRender.Flags().StringVar(&renderOpts.selfHostedEtcdPeerCertPath, "self-hosted-etcd-peer-cert", "", "Path to an existing certificate that will be used for self-hosted etcd's peer port.")
```

Comment:
We already generate certs for external etcd if they are not provided:

https://github.com/kubernetes-incubator/bootkube/blob/master/pkg/asset/tls.go#L118

The functionality should be similar for self-hosted etcd (and shouldn't require many changes).

However, I would like to get this right now, rather than optimizing flags later. It shouldn't be much work. I expect @aaronlevy would agree.

Could you please take a pass at combining flags and making the functionality match was we do for non-self-hosted? It will also help you when implementing tests and updating hack scripts.